File: client-web/src/models/PontoColetaEntrega.js
class PontoColetaEntrega {
    constructor( id, ativaModalEntrega, ehPrimeiroPonto, retornaParaOrigem ) {
      this.id = id;
      this.point = null;
      this.ativaModalEntrega = ativaModalEntrega ?? false;
      this.ehPrimeiroPonto = ehPrimeiroPonto ?? false;
      this.retornaParaOrigem = retornaParaOrigem ?? false;
    }

    setPoint(point) {
        this.point = point;
    }

    setRetornaParaOrigem( retorna ) {
      this.retornaParaOrigem = retorna;
    }
}

export {PontoColetaEntrega};

File: app/lib/model/ponto_coleta_entrega.dart
import 'package:maps_places_autocomplete/model/place.dart';

class PontoColetaEntrega {
  String id;
  Place? place;
  bool ativaModalEntrega;
  bool ehPrimeiroPonto;
  bool retornaParaOrigem;

  PontoColetaEntrega(
      {required this.id,
      this.ativaModalEntrega = false,
      this.ehPrimeiroPonto = false,
      this.retornaParaOrigem = false});

  void setPlace(Place place) {
    this.place = place;
  }

  bool isValidToCotar() {
    return (place != null &&
        place!.lat != null &&
        place!.lng != null &&
        place!.city != null);
  }
}
